src/solvers/refinement/string_functions.h

```diff
@@ -0,0 +1,79 @@
+const irep_idt cprover_string_insert_long_func("__CPROVER_uninterpreted_string_insert_long_func");
+const irep_idt cprover_string_insert_bool_func("__CPROVER_uninterpreted_string_insert_bool_func");
+const irep_idt cprover_string_insert_char_func("__CPROVER_uninterpreted_string_insert_char_func");
+const irep_idt cprover_string_insert_float_func("__CPROVER_uninterpreted_string_insert_float_func");
+const irep_idt cprover_string_insert_double_func("__CPROVER_uninterpreted_string_insert_double_func");
+const irep_idt cprover_string_insert_char_array_func("__CPROVER_uninterpreted_string_insert_char_array_func");
+const irep_idt cprover_string_is_prefix_func("__CPROVER_uninterpreted_string_is_prefix_func");
+const irep_idt cprover_string_is_suffix_func("__CPROVER_uninterpreted_string_is_suffix_func");
+const irep_idt cprover_string_is_empty_func("__CPROVER_uninterpreted_string_is_empty_func");
+const irep_idt cprover_string_last_index_of_func("__CPROVER_uninterpreted_string_last_index_of_func");
+const irep_idt cprover_string_length_func("__CPROVER_uninterpreted_string_length_func");
+const irep_idt cprover_string_data_func("__CPROVER_uninterpreted_string_data_func");
+const irep_idt cprover_string_of_int_func("__CPROVER_uninterpreted_string_of_int_func");
+const irep_idt cprover_string_of_int_hex_func("__CPROVER_uninterpreted_string_of_int_hex_func");
+const irep_idt cprover_string_of_long_func("__CPROVER_uninterpreted_string_of_long_func");
+const irep_idt cprover_string_of_bool_func("__CPROVER_uninterpreted_string_of_bool_func");
+const irep_idt cprover_string_of_float_func("__CPROVER_uninterpreted_string_of_float_func");
+const irep_idt cprover_string_of_double_func("__CPROVER_uninterpreted_string_of_double_func");
+const irep_idt cprover_string_of_char_func("__CPROVER_uninterpreted_string_of_char_func");
+const irep_idt cprover_string_of_char_array_func("__CPROVER_uninterpreted_string_of_char_array_func");
+const irep_idt cprover_string_parse_int_func("__CPROVER_uninterpreted_string_parse_int_func");
+const irep_idt cprover_string_replace_func("__CPROVER_uninterpreted_string_replace_func");
+const irep_idt cprover_string_set_length_func("__CPROVER_uninterpreted_string_set_length_func");
+const irep_idt cprover_string_startswith_func("__CPROVER_uninterpreted_string_startswith_func");
+const irep_idt cprover_string_substring_func("__CPROVER_uninterpreted_string_substring_func");
+const irep_idt cprover_string_to_char_array_func("__CPROVER_uninterpreted_string_to_char_array_func");
+const irep_idt cprover_string_to_lower_case_func("__CPROVER_uninterpreted_string_to_lower_case_func");
+const irep_idt cprover_string_to_upper_case_func("__CPROVER_uninterpreted_string_to_upper_case_func");
+const irep_idt cprover_string_trim_func("__CPROVER_uninterpreted_string_trim_func");
+const irep_idt cprover_string_value_of_func("__CPROVER_uninterpreted_string_value_of_func");
```

Comment:
Rather than define ireps like this, add to the lists built from util/irep*, creating ID_string_trim and so on. Consider replacing "uninterpreted" with just "string_solver" or similar, since the base language string functions aren't being interpreted either.